api/v1alpha1/openapi.yaml

```diff
@@ -3183,3 +3676,97 @@ components:
       - 'TemplateDiscriminatorKubernetesSec'
       - 'TemplateDiscriminatorInlineConfig'
       - 'TemplateDiscriminatorHttpConfig'
+    CertificateSigningRequest:
+      type: object
+      properties:
+        apiVersion:
+          type: string
+          description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+        kind:
+          type: string
+          description: 'Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+        metadata:
+          $ref: '#/components/schemas/ObjectMeta'
+        spec:
+          $ref: '#/components/schemas/CertificateSigningRequestSpec'
+        status:
+          $ref: '#/components/schemas/CertificateSigningRequestStatus'
+      required:
+        - apiVersion
+        - kind
+        - metadata
+        - spec
+      description: 'CertificateSigningRequest represents a request for a signed certificate from the CA'
+    CertificateSigningRequestList:
+      type: object
+      properties:
+        apiVersion:
+          type: string
+          description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+        kind:
+          type: string
+          description: 'Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+        metadata:
+          $ref: '#/components/schemas/ListMeta'
+        items:
+          type: array
+          description: 'List of CertificateSigningRequest.'
+          items:
+            $ref: '#/components/schemas/CertificateSigningRequest'
+      required:
+        - apiVersion
+        - kind
+        - metadata
+        - items
+      description: 'CertificateSigningRequestList is a list of CertificateSigningRequest'
+    CertificateSigningRequestSpec:
+      description: 'Wrapper around a user-created CSR, modeled on kubernetes io.k8s.api.certificates.v1.CertificateSigningRequestSpec'
+      properties:
+        expirationSeconds:
```

Comment:
expiration seconds needs to define type. otherwise the generated go provides an "interface" type

Reply:
Oh yeah, sorry. The schemas commit here is just a placeholder so it can run CI, I'm going to rebase the paths commit onto there once the schemas PR #405  is merged.